rust/operator-binary/src/controller.rs

```diff
@@ -193,6 +207,70 @@ pub async fn reconcile_nifi(nifi: Arc<NifiCluster>, ctx: Arc<Ctx>) -> Result<Act
     tracing::info!("Checking for sensitive key configuration");
     check_or_generate_sensitive_key(client, &nifi).await?;
 
+    // Handle full restarts for a version change
+    let version_change = if let Some(deployed_version) =
+        &nifi.status.clone().unwrap_or_default().deployed_version
```

Comment:
Not prettier but a clone less:
```suggestion
    let version_change = if let Some(deployed_version) = nifi
        .status
        .as_ref()
        .and_then(|status| status.deployed_version.as_ref())
```